app/models/user.rb

```diff
@@ -9,8 +9,29 @@ class User < ApplicationRecord
   accepts_nested_attributes_for :teams
   accepts_nested_attributes_for :team_members
 
-  validates :full_name, presence: true
-  validates :password, length: { minimum: 8 }
+  validates :full_name, presence: true, if: :full_name_needs_validation?
+
+  def full_name_needs_validation?
+    return false if initial_invitation_action?
+
+    return true if profile_update_action?
```

Comment:
could you express the same with
`return [profile_update_action?, accept_invitation_action?].any?`
This will evaluate them in turn and return false if either is false?